### PR TITLE
Copying over all file types under psModules

### PR DESCRIPTION
**Pull request by dsgouda**
## Description

---

This checklist is used to make sure that common guidelines for a pull request are followed.
- [ ] Please add REST spec PR link to the SDK PR
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [ ] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [ ] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [ ] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [ ] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

File: tools/BuildAssets/BuildTasks/Microsoft.Azure.Sdk.Build.Tasks/BootstrapTasks/GetBuildTools.cs [hunk 1]
```diff
@@ -129,10 +130,10 @@ public string RemoteRootDir
                         else
                         {
                             string url = remoteUri.ToString();
-                            if (url.ToLower().Contains("readme.md"))
+                            Regex regex = new Regex("readme.md", RegexOptions.IgnoreCase);
+                            if (regex.IsMatch(url))
```
Comment (shahabhijeet) [inline tools/BuildAssets/BuildTasks/Microsoft.Azure.Sdk.Build.Tasks/BootstrapTasks/GetBuildTools.cs]: @dsgouda you will need to log an error using BuildToolsLogger.LogException/Error incase if the Match isn't successful.
Dump as much data as possible to know what went wrong, as to why we were not able to do a RegEx match.

File: tools/BuildAssets/BuildTasks/Microsoft.Azure.Sdk.Build.Tasks/BootstrapTasks/GetBuildTools.cs [hunk 2]
```diff
@@ -311,6 +312,7 @@ public override bool Execute()
                     {
                         copyFrom = Path.Combine(RemoteCopyFromRootDir, fl);
                         copyTo = Path.Combine(LocalBranchCopyToRootDir, fl);
+                        _buildToolsLogger.LogDebugInfo(string.Format("Copying {0} to {1}", copyFrom, copyTo));
```
Comment (shahabhijeet) [inline tools/BuildAssets/BuildTasks/Microsoft.Azure.Sdk.Build.Tasks/BootstrapTasks/GetBuildTools.cs]: @dsgouda use the actual property name, do not use the field for BuildToolsLogger

File: tools/BuildAssets/BuildTasks/Microsoft.Azure.Sdk.Build.Tasks/BootstrapTasks/GetBuildTools.cs [hunk 3]
```diff
@@ -379,12 +380,12 @@ private void CopyPowershellModules(string psModulesDir, string[] filesToCopy)
...
             string userProfilePSModulesPath = new Uri(Environment.GetEnvironmentVariable("USERPROFILE") + "\\Documents\\WindowsPowerShell\\Modules").AbsolutePath;
             
-            IEnumerable<string> psModulesToCopy = filesToCopy.Where(p=>p.StartsWith("psModules"))
-                                                             .Where(p=>p.EndsWith(".psm1")||p.EndsWith(".psd1")||p.EndsWith(".ps1"));
+            IEnumerable<string> psModulesToCopy = filesToCopy.Where(p=>p.StartsWith("psModules"));
             foreach (var module in psModulesToCopy)
             {
                 var srcFile = Path.GetFullPath(Path.Combine(psModulesDir, module));
                 var destFile = Path.GetFullPath(Path.Combine(userProfilePSModulesPath, module.Replace("psModules/", "")));
+                _buildToolsLogger.LogDebugInfo(string.Format("Copying {0} to {1}", srcFile, destFile));
```
Comment (shahabhijeet) [inline tools/BuildAssets/BuildTasks/Microsoft.Azure.Sdk.Build.Tasks/BootstrapTasks/GetBuildTools.cs]: @dsgouda use BuildToolsLogger

**Review by shahabhijeet — CHANGES_REQUESTED** [review]

File: tools/BuildAssets/BuildTasks/Microsoft.Azure.Sdk.Build.Tasks/BootstrapTasks/GetBuildTools.cs [hunk 4]
```diff
@@ -129,10 +130,14 @@ public string RemoteRootDir
...
+                            Regex regex = new Regex("readme.md", RegexOptions.IgnoreCase);
+                            if (regex.IsMatch(url))
                             {
-                                url = url.ToLower().Replace("readme.md", "");
-                                _remoteRootDir = url;
+                                _remoteRootDir = regex.Replace(url, "");
+                            }
+                            else
+                            {
+                                BuildToolsLogger.LogException(new Exception(string.Format("Url {0} does not contain readme.md", url)));
```
Comment (shahabhijeet) [inline tools/BuildAssets/BuildTasks/Microsoft.Azure.Sdk.Build.Tasks/BootstrapTasks/GetBuildTools.cs]: @dsgouda add a test for this scenario, this way we know this code path works the way we want it to work.

**Review by shahabhijeet — CHANGES_REQUESTED** [review]
Once you add the test, and CI passes, this is good to be merged.

File: tools/BuildAssets/BuildTasks/Microsoft.Azure.Sdk.Build.Tasks/BootstrapTasks/GetBuildTools.cs (hunk 4 above)
Comment (dsgouda) [reply tools/BuildAssets/BuildTasks/Microsoft.Azure.Sdk.Build.Tasks/BootstrapTasks/GetBuildTools.cs]: Good catch.

**Review by shahabhijeet — APPROVED** [review]

**Pull request merged**